moveit_ros/moveit_servo/src/servo_calcs.cpp

```diff
@@ -576,7 +595,53 @@ bool ServoCalcs::cartesianServoCalcs(geometry_msgs::msg::TwistStamped& cmd,
   Eigen::MatrixXd matrix_s = svd.singularValues().asDiagonal();
   Eigen::MatrixXd pseudo_inverse = svd.matrixV() * matrix_s.inverse() * svd.matrixU().transpose();
 
-  delta_theta_ = pseudo_inverse * delta_x;
+  // Convert from cartesian commands to joint commands
+  // Use an IK solver plugin if we have one, otherwise use inverse Jacobian.
+  if (!use_inv_jacobian_)
+  {
+    // get a transformation matrix with the desired position change &
+    // get a transformation matrix with desired orientation change
+    Eigen::Isometry3d tf_pos_delta(Eigen::Isometry3d::Identity());
+    tf_pos_delta.translate(Eigen::Vector3d(delta_x[0], delta_x[1], delta_x[2]));
+
+    Eigen::Isometry3d tf_rot_delta(Eigen::Isometry3d::Identity());
+    Eigen::Quaterniond q = Eigen::AngleAxisd(delta_x[3], Eigen::Vector3d::UnitX()) *
+                           Eigen::AngleAxisd(delta_x[4], Eigen::Vector3d::UnitY()) *
+                           Eigen::AngleAxisd(delta_x[5], Eigen::Vector3d::UnitZ());
+    tf_rot_delta.rotate(q);
+
+    // Poses passed to IK solvers are assumed to be in the tip link (EE) reference frame
+    // So, apply the base->end effector transform to the delta transform
+    // apply rotation after so that it's local to EE
+    auto full_tf = tf_pos_delta * tf_moveit_to_ee_frame_ * tf_rot_delta;
+    geometry_msgs::msg::Pose next_pose = tf2::toMsg(full_tf);
+
+    // setup for IK call
+    static std::vector<double> solution(num_joints_, 0.0);
+    static moveit_msgs::msg::MoveItErrorCodes err;
+    static kinematics::KinematicsQueryOptions opts;
```

Comment:
Sorry to backtrack. From reading here, `static` does not really help in terms of realtime performance. Please remove `static`.  :P

https://stackoverflow.com/a/3835902